flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -464,7 +459,13 @@ public void failJobDueToTaskFailure(
                         checkpointsCleaner,
                         new ScheduledExecutorServiceAdapter(checkpointCoordinatorTimer),
                         SharedStateRegistry.DEFAULT_FACTORY,
-                        failureManager);
+                        failureManager,
+                        new CheckpointBriefCalculator(
+                                getJobID(),
+                                sourceAndAllVertices.f0,
+                                sourceAndAllVertices.f1,
+                                sourceAndAllVertices.f1),
+                        new ExecutionAttemptMappingProvider(sourceAndAllVertices.f1));
```

Comment:
I think `ExecutionGraph` needs `getSourceAndAllVertices()` only to construct these two objects.
Why not move this method to `CheckpointBriefCalculator`. 
I guess it will also make future steps easier (dynamic sources calculation).

`ExecutionAttemptMappingProvider` can then depend on `CheckpointBriefCalculator`.

WDYT?

Reply:
I also more like to move the creation to be inside `CheckpointPlanCalculator`. However, since currently the unit tests relies on passing mock vertices directly on creating `CheckpointCoordinator`, thus I think we may postpone the change to the next PR, in the second PR we would refactor the current tests to be based on real `ExecutionGraph` and then we could move the creation into inside `CheckpointPlanCalculator`.

I'm a little concern for make `ExecutionAttemptMappingProvider` to be depend on `CheckpointPlanCalculator` since logically I think the two components should be for two separate purposes. Besides, we would also adjust the logic of the `CheckpointPlanCalculator` in the next PR. Do you think it would be also ok to separate the two components and make them all depends on ExecutionGraph~?

Reply:
I share your concern and I also like your proposal :+1: